src/example/DistributedPhysical/Distributed_dsp.ts

```diff
@@ -0,0 +1,142 @@
+// Code generated by the Lingua Franca compiler from file:
```

Comment:
Generally, we should not be checking in generated code, but I see why you'd want to have this readily available in the example directory...

Reply:
Yes, normally I would not check in generated code.

The examples are here because I wanted to make it possible to test modifications to the reactor-ts submodule without resolving my pull request to the LF repo. The code isn't 100% generated either. I started with generated code, but I modified these by hand to make them as simple as possible to work with.

Reply:
I actually think these examples are best put in the `lingua-franca` repo, not the `reactor-ts` repo. In that case we also don't have to check in generated code, which is generally considered a bad practice.

This also makes it abundantly clear that the expected workflow for writing federated reactors is to use the compiler tool chain.

In short, I think that examples in reactor-ts should feature hand-written reactors only.

Reply:
I have a different example program checked into the LF repo (which is a .lf file), so I don't think we should move this there.

Although these examples began life as generated code, they have been significantly modified from that by hand. To me, this qualifies them as hand-written. If I were to further clean them up so they didn't have any of the code-generated boilerplate at all, would that make them check-in-able to you?

> This also makes it abundantly clear that the expected workflow for writing federated reactors is to use the compiler tool chain.

Why is this the expected workflow? It doesn't seem particularly more difficult to me to write a federated reactor by hand in TypeScript than it is to write a moderately complex non-federated reactor by hand in TypeScript. There are something like 5 extra functions you have to use to set it up.

Reply:
I understand that generated code can be polished up to look like it was handwritten, but that doesn't make them handwritten. If this is a different example than the one in the LF repo, then why not just have two examples in the LF repo?

Having to figure out numerical federate IDs and port IDs in order to plug things together strikes me as brittle and difficult to write an API manual for. This API was written for a code generator, not for a programmer. I think we should write an API that is accessible for programmers, and until we do, I suggest not advertising this as a feature of reactor-ts. Like I stated before, this approach works fine in LF, because there all the marshaling between reactors is abstracted using ordinary connections. This is why I say using the LF compiler is the expected workflow.

Reply:
> I understand that generated code can be polished up to look like it was handwritten, but that doesn't make it handwritten. If this is a different example than the one in the LF repo, then why not just have two examples in the LF repo?

Because these examples illustrate how you can write a federated program in TypeScript. They make it possible to try out changes to how federates work in TypeScript without having to rebuild the LF compiler. Would it satisfy your desire to not advertise these as a feature of reactorTS if I added a README into their directory, explaining that point?

> Having to figure out numerical federate IDs and port IDs in order to plug things together strikes me as brittle and difficult to write an API manual for.

To me this task is pretty similar to working with threadIDs or processIDs. Would an API manual really be that bad? "Assign each federatedApp an ID in its constructor. The ID must be unique among all federates and be a number between 0 and NUMBER_OF_FEDERATES - 1"